### PR TITLE
[INFINITY-1054] Fix suppression by only considering running plans whe…

### PR DESCRIPTION
**Pull request by mgummelt**
…n deciding when to suppress

Blocked on:
#751 
#752 
#753 
#754

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,6 +114,15 @@ volume:
...
   size: 64
 ```
+===============
+### Testing
+
+```bash
+$ virtualenv -p python3 venv
+$ source venv/bin/activate
+$ pip install -r testing/requirements.txt
+$ py.test frameworks/helloworld/tests
```
Comment (gabrielhartmann) [inline README.md]: `CLUSTER_URL=https://gabriel-4-elasticl-1fw1wa1b32i2d-307940316.us-west-2.elb.amazonaws.com/#/settings/repositories?_k=eziook TEST_TYPES="sanity" ./test.sh helloworld`

Comment (gabrielhartmann) [inline README.md]: This should also not be in this PR.

Comment (mgummelt) [inline README.md]: `test.sh` includes wrapper features that aren't relevant to developers (CCM, jenkins, etc.).  `py.test` is a standard that people are familiar with.  We shouldn't introduce a non-standard tool unless it provides valuable features.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/Element.java [hunk 2]
```diff
@@ -98,6 +98,11 @@ default boolean isComplete() {
     }
 
     /**
+     * Indicates whether this Element is running
```
Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/Element.java]: Missing trailing `.`

File: README.md (hunk 1 above)
Comment (gabrielhartmann) [inline README.md]: If you want to change all the test documentation, please do so in another PR.

Comment (mgummelt) [inline README.md]: removed

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/Element.java (hunk 2 above)
Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/Element.java]: fixed

File: testing/sdk_install.py [hunk 3]
```diff
@@ -22,31 +22,36 @@ def install(
...
     app_id = sdk_marathon.get_config(service_name)['id']
     marathon_client = dcos.marathon.create_client()
     def fn():
         # TODO(nickbp): upstream fix to shakedown, which currently checks for ANY deployments rather
         #               than the one we care about
         deploying_apps = set([])
-        for d in marathon_client.get_deployments():
-            for a in d.get('affectedApps', []):
-                deploying_apps.add(a)
+        for deployment in marathon_client.get_deployments():
```
Comment (gabrielhartmann) [inline testing/sdk_install.py]: Was the intention to also add a check that a framework is suppressed once install is complete?

Comment (mgummelt) [inline testing/sdk_install.py]: Yes, see the description.

**Comment by mgummelt** [comment]
@gabrielhartmann See the description.  I can't check suppression state from an integration test.

**Comment by nickbp** [comment]
PR 589 has a fix for the caching issue and exercises the property retrieval in an integration test: https://github.com/mesosphere/dcos-commons/pull/589/files#diff-eb978dabd90a8fa48c742f28b2b56c14R149

**Comment by mgummelt** [comment]
@gabrielhartmann Updated with an integration test, and ready for review.

**Comment by mgummelt** [comment]
retest this please

**Comment by mgummelt** [comment]
retest this please

**Comment by mgummelt** [comment]
retest this please

**Comment by mgummelt** [comment]
cc @gabrielhartmann tests are passing. ready for review.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultPlanCoordinator.java [hunk 4]
```diff
@@ -99,7 +99,14 @@ public DefaultPlanCoordinator(
 
     @Override
     public boolean hasOperations() {
-        return planManagers.stream().anyMatch(manager -> !manager.getPlan().isComplete());
+        boolean ret = false;
+        for (final PlanManager planManager : planManagers) {
```
Comment (mrbrowning) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultPlanCoordinator.java]: Is this expansion into `for` for the sake of adding logging?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java [hunk 5]
```diff
@@ -219,11 +219,15 @@ public static void validateValue(byte[] value) throws StateStoreException {
     public static boolean isSuppressed(StateStore stateStore) throws StateStoreException {
         byte[] bytes = fetchPropertyOrEmptyArray(stateStore, SUPPRESSED_PROPERTY_KEY);
 
-        try {
-            return new JsonSerializer().deserialize(bytes, Boolean.class);
-        } catch (IOException e){
-            LOGGER.error(String.format("Error converting property '%s' to boolean.", SUPPRESSED_PROPERTY_KEY), e);
-            throw new StateStoreException(Reason.SERIALIZATION_ERROR, e);
+        if (bytes.length == 0) {
```
Comment (mrbrowning) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java]: I can understand shortcutting for the trivial case (and being averse to exception-as-control-flow), but does it really buy us anything right now over the slightly more readable previous formulation?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultPlanCoordinator.java (hunk 4 above)
Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultPlanCoordinator.java]: Yea, I found it quite useful during debugging.  Not that there's any way to enable DEBUG logging currently...

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java (hunk 5 above)
Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java]: It was buggy before.  If the key wasn't set, the scheduler would throw the exception and die.  This makes it so that not set -> false, which is what we want.

**Comment by mgummelt** [comment]
Thanks @mrbrowning
